src/shared/utils/windowPeriod/fromDuration.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  SELECTABLE_TIME_RANGES,
+  DESIRED_POINTS_PER_GRAPH,
+} from 'src/shared/constants/timeRanges'
+
+function calcWindowPeriodForRangeDuration(rangeDuration: number) {
+  return Math.round(rangeDuration / DESIRED_POINTS_PER_GRAPH) // duration in ms
+}
+
+/**
+ * Determining the `windowPeriod`
+ * from the timeRange numerical duration (a.k.a. upper - lower) in milliseconds.
+ *
+ * @param rangeDuration -- given the duration in milliseconds
+ * @returns {number} -- numeric value of windowPeriod
+ */
+export function convertMillisecondDurationToWindowPeriod(
+  rangeDuration: number
+): number {
+  const foundDuration = SELECTABLE_TIME_RANGES.find(
+    tr => tr.seconds * 1000 === rangeDuration
+  )
+  if (foundDuration) {
+    return foundDuration.windowPeriod
```

Comment:
we set the timeRange that we want, per the DatePicker, in `SELECTABLE_TIME_RANGES`. It may be override by the user in the internal flux query (which the visitor then finds). But we do want the the default window period is constant.

How does this method `convertMillisecondDurationToWindowPeriod()` get used? It makes sure that the default windowPeriod per timeRange, is same in both of these cases:

* Scenario 1:  DatePicker has -15m
   ```
   sample.data(set: "airSensor")
        |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
   ```
* Scenario 2:  user writes range for -15m
   ```
   sample.data(set: "airSensor")
        |> range(start: -15m, stop: now())
   ```

Reply:
@asalem1 - Here the logic is return a preset windowPeriod if the time range is non-custom, OR if custom range is used, then calculate the windowPeriod (see calcWindowPeriodForRangeDuration below).  The function below does the range / default number of points (360).